python/pyarrow/tests/test_pandas.py

```diff
@@ -2860,9 +2852,8 @@ def _fully_loaded_dataframe_example():
         9: pd.period_range('2013', periods=10, freq='M')
     }
 
-    if Version(pd.__version__) >= Version('0.21'):
-        # There is an issue with pickling IntervalIndex in pandas 0.20.x
-        data[10] = pd.interval_range(start=1, freq=1, periods=10)
+    # There is an issue with pickling IntervalIndex in pandas 0.20.x
+    data[10] = pd.interval_range(start=1, freq=1, periods=10)
```

Comment:
I think you can remove the comment as well (that was only relevant when putting it behind the `if` version check). 
And then you can also move that inside the dict above (`10: pd.internval_range ...`)